pkg/common/common.go

```diff
@@ -172,19 +172,19 @@ func MarshallNamespace(gwKey client.ObjectKey, domain string) string {
 }
 
 func UnMarshallObjectKey(keyStr string) (client.ObjectKey, error) {
-	keySplit := strings.Split(keyStr, string(NamespaceSeparator))
-	if len(keySplit) < 2 {
-		return client.ObjectKey{}, fmt.Errorf("failed to split on %s: '%s'", string(NamespaceSeparator), keyStr)
+	namespaceEndIndex := strings.IndexByte(keyStr, NamespaceSeparator)
+	if namespaceEndIndex < 0 || len(keyStr)-namespaceEndIndex-1 < 1 {
```

Comment:
The function is returning an error if the `NamespaceSeparator` is not found or it is the last char. 

What if it is the first one? I think it is enough to know there is at least one and then read both sides of the first separator. Which, btw, any of them could be empty, yes.

Reply:
I see! I added two test-cases for that and going to push them now to the PR (https://github.com/Kuadrant/kuadrant-operator/pull/186):
1. when valid namespace with empty object (ends with '/') then return ObjectKey with namespace only
2. when valid name with empty namespace (starts with '/') then return ObjectKey with name only

Regarading changes, I think, this simplification of the condition includes the cases you described:

```
if namespaceEndIndex < 0 {
    return client.ObjectKey{}, fmt.Errorf(fmt.Sprintf("failed to split on %s: '%s'", string(NamespaceSeparator), keyStr))
}
```

This implementation passes the unit-tests, which are in the PR (https://github.com/Kuadrant/kuadrant-operator/pull/186) + two new test-cases mentioned

What do you think @eguzki?

Reply:
I made the changes and pushed them here: https://github.com/Kuadrant/kuadrant-operator/pull/187/commits/159b206bf9820b3aa4d8d4a23a2f41d83621a7f5

Thanks again, @eguzki :angel: